app/controllers/v0/form21p530a_controller.rb

```diff
@@ -73,5 +89,34 @@ def handle_pdf_generation_error(error)
     def stats_key
       'api.form21p530a'
     end
+
+    def transform_country_codes(payload)
+      parsed = JSON.parse(payload)
+      address = parsed.dig('burialInformation', 'recipientOrganization', 'address')
+      if address&.key?('country')
+        transformed_country = extract_country(address)
+        if transformed_country
+          # Validate that the transformed country code is a valid ISO code
+          # This prevents invalid codes from making it onto the PDF for both create and download_pdf
+          validate_country_code!(transformed_country)
+          address['country'] = transformed_country
+        end
+      end
+      parsed.to_json
+    end
+
+    def validate_country_code!(country_code)
+      return if country_code.blank?
+
+      # Verify it's a valid ISO 3166-1 Alpha-2 code
+      IsoCountryCodes.find(country_code)
+    rescue IsoCountryCodes::UnknownCodeError
+      # Create a temporary claim object to hold the validation error
+      # This allows us to use ValidationErrors exception which expects an ActiveModel object
+      claim = SavedClaim::Form21p530a.new
+      claim.errors.add '/burialInformation/recipientOrganization/address/country',
+                       "'#{country_code}' is not a valid country code"
+      raise Common::Exceptions::ValidationErrors, claim
+    end
```

Comment:
These ultimately need to be moved out of the controller and refactored - likely into the model or a validator of some sort. Out of scope for this effort, but this will need to be done eventually.

Created a ticket to do this eventually: https://github.com/department-of-veterans-affairs/va.gov-team/issues/124779